lncfg/db.go

```diff
@@ -50,12 +51,14 @@ func (db *DB) Validate() error {
 }
 
 // GetBackend returns a kvdb.Backend as set in the DB config.
-func (db *DB) GetBackend(path string) (kvdb.Backend, error) {
+func (db *DB) GetBackend(dbPath string) (kvdb.Backend, error) {
 	if db.Backend == etcdBackend {
-		return kvdb.GetEtcdBackend(db.Etcd)
+		// Prefix will separate key/values in the db.
+		prefix := path.Join(dbPath, dbName)
```

Comment:
We can shorten this a bit, by just extracting which _network_ the user is running on and using that as a prefix. Otherwise, we'll end up writing part of the user's file system path as the main bucket key for everything, which can possibly change depending on where they're running `lnd` and what they have set as the `datadir`, etc. 

The other item we should tack on here is the pubkey of the node, which is known before we update the database. So the final prefix would be: `/network/<nodePubKey/`.

Reply:
This is not an actual prefix, perhaps the naming is not the best as it confuses readers :). It is the same as if there were an extra bucket, which means that in the end the whole path (except the key) is flattened to 32 bytes.

Reply:
So the "flattening" (key generation?) algorithm is a bit easier to grasp if you take a look at `bucket_test.go` and how the `bkey` function is used in the tests. The way it is currently done, we do create the key when we create a `ReadWriteBucket` and just use the cached key after that. Each bucket key is prefixed with a byte containing `b` and each value key is prefixed with a byte containing `v` (for details please look at: `bucket.go`.

Reply:
i like the idea of not prefixing all keys with redundant data, having a root bucket as H(network | nodepub) make sense 👍

it does seem like that path in full should be removed, otherwise one could have multiple states for the same (node, network) pair which is dangerous